From c366b643d7feecdeb4ce5e78e72f090199aed931 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Mon, 20 Jun 2016 21:14:33 +0800
Subject: [PATCH 1216/1240] spl: a3700: stop building SPL

- SPL is no longer part of Armada3700 ATF boot image, though it is
  still compiled in u-boot.
- Update defconfig for Armada3700 to stop building SPL
- Remove spl image variable - "SPLIMAGE" in Makefile
- Remove oslo start addr option in kconfig file since only armada 3700
  spl uses it

Change-Id: I23a2dac6373baf373275eea1d11cb46708cc9f3d
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30440
Reviewed-by: Hua Jing <jinghua@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 Makefile                              |  1 -
 arch/arm/cpu/armv8/armada3700/Kconfig | 10 ----------
 configs/mvebu_armada3700_defconfig    | 16 +++++++---------
 3 files changed, 7 insertions(+), 20 deletions(-)

diff --git a/Makefile b/Makefile
index 5100f82..e5a1f94 100644
--- a/Makefile
+++ b/Makefile
@@ -838,7 +838,6 @@ u-boot.ldr.hex u-boot.ldr.srec: u-boot.ldr FORCE
 ifdef CONFIG_MVEBU
 ifeq ($(CONFIG_TARGET_ARMADA_3700), y)
 
-SPLIMAGE	:= $(srctree)/spl/u-boot-spl.bin
 BIN2PHEX	:= $(srctree)/scripts/bin2phex.pl
 
 endif # CONFIG_TARGET_ARMADA_3700
diff --git a/arch/arm/cpu/armv8/armada3700/Kconfig b/arch/arm/cpu/armv8/armada3700/Kconfig
index 4e5f07a..fdebbd1 100644
--- a/arch/arm/cpu/armv8/armada3700/Kconfig
+++ b/arch/arm/cpu/armv8/armada3700/Kconfig
@@ -24,15 +24,5 @@ config ARMV8_PSCI
 	bool
 	default y
 
-config OSLO_START_ADDR
-        hex "OSLO image start address"
-        default 4100000
-        help
-          Define the OSLO image start address.
-          In case of SPL+u-boot, it is the u-boot start address 0,
-          in case of SPI+ATF+u-boot, it is the start
-          address of ATF.
-          The default OSLO start address 0x4100000 is the ATF image start
-          address after it is loaded.
 
 endif
diff --git a/configs/mvebu_armada3700_defconfig b/configs/mvebu_armada3700_defconfig
index b0c0fab..976f54a 100644
--- a/configs/mvebu_armada3700_defconfig
+++ b/configs/mvebu_armada3700_defconfig
@@ -1,11 +1,10 @@
-CONFIG_SPL=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
-+S:CONFIG_ARM=y
-+S:CONFIG_TARGET_ARMADA_3700=y
+CONFIG_ARM=y
+CONFIG_TARGET_ARMADA_3700=y
 CONFIG_MVEBU_A3700_IO_ADDR_DEC=y
 CONFIG_MVEBU_A3700_CLOCK=y
 CONFIG_MVEBU_MPP_BUS=y
-+S:CONFIG_MVEBU_A3700_UART=y
+CONFIG_MVEBU_A3700_UART=y
 CONFIG_XENON_MMC=y
 CONFIG_DEVEL_BOARD=y
 CONFIG_CMD_BDI=y
@@ -24,9 +23,9 @@ CONFIG_CMD_MVEBU_MPP=y
 CONFIG_CMD_MVEBU_FDT_CONFIG=y
 CONFIG_CMD_MVEBU_BUBT=y
 CONFIG_CMD_MVEBU_HW_INFO=y
-+S:CONFIG_OF_CONTROL=y
-+S:CONFIG_OF_EMBED=y
-+S:CONFIG_DEFAULT_DEVICE_TREE="armada-3700-db"
+CONFIG_OF_CONTROL=y
+CONFIG_OF_EMBED=y
+CONFIG_DEFAULT_DEVICE_TREE="armada-3700-db"
 CONFIG_FDT_SIZE=0x1C00
 CONFIG_DM=y
 CONFIG_MVEBU_ADVK_PCIE=y
@@ -44,9 +43,8 @@ CONFIG_MVEBU_GPIO=y
 CONFIG_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_MVEBU_COMPHY_SUPPORT=y
-+S:# CONFIG_MVEBU_SPL_DDR_SUPPORT is not set
 CONFIG_SHA256=y
 CONFIG_SHA1=y
 CONFIG_MVEBU_BOOTMODE_SWITCH_SUPPORT=y
-+S:CONFIG_MVEBU_MBUS_SKIP_DRAM_WIN=y
+CONFIG_MVEBU_MBUS_SKIP_DRAM_WIN=y
 CONFIG_MVEBU_A3700_PM=y
-- 
1.9.1

